client/packages/system/src/Patient/PatientView/PatientView.tsx

```diff
@@ -164,7 +165,13 @@ const PatientDetailView = ({
           isDeceased: currentPatient.isDeceased ?? undefined,
           phone: currentPatient.phone ?? undefined,
           address1: currentPatient.address1 ?? undefined,
-          nextOfKinId: currentPatient.nextOfKinId ?? undefined,
+          nextOfKin:
+            currentPatient.nextOfKinId || currentPatient.nextOfKinName
+              ? {
+                  id: currentPatient.nextOfKinId ?? undefined,
+                  name: currentPatient.nextOfKinName ?? undefined,
```

Comment:
small bit of ick - works best for JSON schema to have nextOfKin as an object, but it needs to be a _different_ nextOfKin object than the `nextOfKin` field on `currentPatient`, which is a whole PatientNode from the backend, only if the nextOfKinId is set. 

Found the best was just to query the two distinct fields from API, and build the JSON form nextOfKin object here, then map back to separate fields on save 🥲